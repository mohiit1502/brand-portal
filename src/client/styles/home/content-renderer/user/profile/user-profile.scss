@import "../../../../override/variables";

.user-profile-content {
  .content-header-row {
    font-weight: bold;
  }
  .password-reset-col{
    button{
      border-radius: 42px;
      font-weight: bold;
      font-size: 15px;

      &.reset-password {
        border: $default-input-border;
        color: $market-place-blue;
      }

      &.manage-profile {
        color: white;
      }

      &:hover {
        color: white;
        background: $market-place-blue;
      }
    }
  }

  .edit-button-panel {
    .btn-primary{
      border-radius: 42px;
      background: #006799;
      font-weight: bold;
      font-size: 15px;
      border: none;
    }
    .btn-link{
      font-weight: bold;
      font-size: 18px;
      text-decoration: underline;
    }
  }
  .password-reset-row {
    .badge {
      //border: 1px solid $blue;
    }
  }
}
