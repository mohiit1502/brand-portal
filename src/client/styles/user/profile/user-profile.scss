@import "../../override/variables";

.user-profile-content {
  .password-reset-row {
    .badge {
      border: 1px solid $blue;
    }
  }
}
