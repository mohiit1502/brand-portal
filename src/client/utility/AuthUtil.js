import authConfig from "./../config/authorizations";
import Helper from "./helper";

export default class AuthUtil {
  static isActionAccessible (action, userProfile, actedOnRole) {
    if (action && action.indexOf(".") > -1) {
      const actionConfig = Helper.search(action, authConfig);
      if (actionConfig && userProfile && userProfile.role && userProfile.role.name && typeof actionConfig === "object") {
        const role = userProfile.role.name;
        if (actionConfig.length !== undefined) {
          return actionConfig.includes(role);
        } else {
          if (Object.keys(actionConfig).includes(role) && actedOnRole && actionConfig[role].includes(actedOnRole)) {
            return true;
          }
        }
      }
    }
    return false;
  }
}
