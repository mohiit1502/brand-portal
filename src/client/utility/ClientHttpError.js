export default class ClientHttpError {

  status;
  error;
  message;

  constructor (status, error, message) {
    this.status = status;
    this.error = error;
    this.message = message;
  }

}
