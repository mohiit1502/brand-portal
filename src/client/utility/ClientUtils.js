import queryString from "query-string";

export default class ClientUtils {

  static getQueryParams (location) {
    return queryString.parse(location.search);
  }
}
