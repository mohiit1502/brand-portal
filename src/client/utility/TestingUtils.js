import {applyMiddleware, compose, createStore} from "redux";
import rootReducer from "../reducers";
import thunk from "redux-thunk";
import Immutable from "immutable";
import configureMockStore from 'redux-mock-store';
import ClientHttpError from "./ClientHttpError";

export const findByTestAttribute = (component,attr) => component.find(`[data-test="${attr}"]`);

export const testStore = state => configureMockStore()(state);

export const realStore = state => {
  const configureStore = initialState => {
    const composeEnhancers =
      typeof window === "object" &&
      window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ ?
        window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({
          serialize: { // prettier-ignore
            immutable: Immutable
          }
        }) : compose;
    const store = createStore(rootReducer, initialState, composeEnhancers(applyMiddleware(thunk)));

    if (module.hot) {
      module.hot.accept("./reducers", () => {
        const nextRootReducer = require("./reducers").default;
        store.replaceReducer(nextRootReducer);
      });
    }

    return store;
  };

  return configureStore(state || window.__WML_REDUX_INITIAL_STATE__);
}

export const clearKeys = (tree, arr) => {
  if (arr && arr.indexOf(tree) > -1) {
    return;
  }
  if (tree && typeof tree === 'object') {
    if (tree.inputId) {
      return;
    }
    arr.push(tree)
    if (tree.key) {
      try {
        delete tree.key;
      } catch (e) {}
    }
    if (Array.isArray(tree)) {
      tree.forEach(obj => clearKeys(obj, arr));
    } else {
      Object.keys(tree).forEach(key => clearKeys(tree[key], arr));
    }
  }
}

export function setupFetchStub(data) {
  return function fetchStub(_url) {
    return new Promise((resolve) => {
      resolve({
        json: () =>
          Promise.resolve({
            data,
          }),
      })
    })
  }
}

export function setupFetchThrowStub() {
  const error = new ClientHttpError(500, "test-error"); // Directly throwing new ClientHttpError fails in Jest
  throw error;
}
