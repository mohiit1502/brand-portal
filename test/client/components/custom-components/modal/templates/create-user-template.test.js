/* eslint-disable filenames/match-regex, no-unused-vars, no-undef */
import "core-js/stable";
import "regenerator-runtime/runtime";
import React from "react";
import {Provider} from "react-redux";
import renderer, {act} from "react-test-renderer";
import {mount} from "enzyme";
import Cookies from "electrode-cookies";
import {clearFetchMock, mockFailFetch, mockFetch, testStore} from "../../../../../../src/client/utility/TestingUtils";
import CreateUserTemplate from "../../../../../../src/client/components/custom-components/modal/templates/create-user-template";
import MODALSMETA from "../../../../../../src/client/config/contentDescriptors/modals-meta";
import FORMFIELDMETA from "../../../../../../src/client/config/formsConfig/form-field-meta";
import profile from "../../../../mocks/userProfile";
import CONSTANTS from "../../../../../../src/client/constants/constants";

let store;
let root;
let data;
jest.mock("electrode-cookies");

const setUpRender = () => {
  const mockStore = {
    content: {
      metadata: {...MODALSMETA, ...FORMFIELDMETA}
    },
    user: {profile}
  };
  store = testStore(mockStore);
  act(() => {
    root = renderer.create(<Provider store={store}><CreateUserTemplate/></Provider>);
  });
  return root;
};

const setUpMount = () => {
  const mockStore = {
    content: {
      metadata: {...MODALSMETA, ...FORMFIELDMETA}
    },
    user: {profile}
  };
  store = testStore(mockStore);
  data = profile;
  return mount(<Provider store={store}><CreateUserTemplate data={data}/></Provider>);
};

describe("CreateUserTemplate test container", () => {
  let wrapper;
  beforeEach(() => {
    jest.restoreAllMocks();
    jest.resetAllMocks();
    Cookies.get.mockImplementation(jest.fn());
  });

  it("should render the CreateUserTemplate successfully", () => {
    wrapper = setUpRender();
    mockFetch({
      data: {
        shouldMap: true,
        "/api/newUser/roles": {roles: [{name: "test-role"}]},
        "/api/newUser/brands": {brands: [{brandName: "test-brand"}]}
      }
    });
    expect(wrapper.toJSON()).toMatchSnapshot();
    act(() => {
      root.update(<Provider store={store}><CreateUserTemplate data={profile}/></Provider>);
    });
    expect(wrapper.toJSON()).toMatchSnapshot();
    wrapper.root.findByProps({className: "close text-white"}).props.onClick();
  });

  it("should trigger the new user workflow successfully", async () => {
    jest.useFakeTimers();
    mockFetch({
      data: {
        shouldMap: true,
        "/api/newUser/roles": {roles: [{name: "Super Admin"}, {name: "Administrator"}, {name: "Reporter"}]},
        "/api/newUser/brands": {brands: [{name: "test-brand", brandName: "test-brand", id: "test-brand"}, {name: "all", brandName: "all", id: "all"}]},
        "/api/users": {brands: [{name: "test-brand", brandName: "test-brand", id: "test-brand"}, {name: "all", brandName: "all", id: "all"}]}
      }
    });
    let wrapper = setUpMount();
    await Promise.resolve();
    await Promise.resolve();
    await Promise.resolve();
    await Promise.resolve();
    wrapper.update();
    wrapper.find(".close.text-white").at(0).simulate('click');
    wrapper.find(".custom-control-input").at(0).simulate('change');
    wrapper.find(".form-control-emailId").at(0).simulate('change');
    wrapper.find(".form-control-phone").at(0).simulate('change');
    wrapper.find(".form-control-phone").at(0).simulate('change', {target: {value: "56dx", checkValidity: jest.fn()}});
    wrapper.find(".dropdown-item").at(0).simulate('click');
    wrapper.find({type: "checkbox", className: "cursor-pointer my-auto"}).at(0).simulate('change');
    wrapper.find("form.h-100.px-2.pl-0").at(0).simulate('submit');
    jest.runAllTimers();
    jest.useRealTimers();
  });

  it("should trigger the new user workflow and fail", async () => {
    mockFetch({
      data: {
        shouldMap: true,
        "/api/newUser/roles": {roles: [{name: "Super Admin"}, {name: "Administrator"}, {name: "Reporter"}]},
        "/api/newUser/brands": {brands: [{name: "test-role", brandName: "test-role", id: "test-role"}, {name: "all", brandName: "all", id: "all"}]},
        "/api/users": undefined
      }
    });
    const mockStore = {
      content: {metadata: {...MODALSMETA, ...FORMFIELDMETA}},
      user: {profile}
    };
    store = testStore(mockStore);
    wrapper = mount(<Provider store={store}><CreateUserTemplate /></Provider>);
    wrapper.find(".dropdown-item").at(0).simulate('click');
    wrapper.update()
    wrapper.find("form.h-100.px-2.pl-0").at(0).simulate('submit');
  });

  it("should trigger the update workflow with failure", async () => {
    const formFieldMeta = JSON.parse(JSON.stringify(FORMFIELDMETA));
    formFieldMeta.FORMSCONFIG.NEWUSER.formConfig.isUpdateTemplate = true;
    mockFailFetch();
    const mockStore = {
      content: {metadata: {...MODALSMETA, ...formFieldMeta}},
      user: {profile}
    };
    store = testStore(mockStore);
    wrapper = mount(<Provider store={store}><CreateUserTemplate data={data}/></Provider>);
    wrapper.find("form.h-100.px-2.pl-0").at(0).simulate('submit');
  });

  it("should trigger the update workflow", async () => {
    const formFieldMeta = JSON.parse(JSON.stringify(FORMFIELDMETA));
    formFieldMeta.FORMSCONFIG.NEWUSER.formConfig.isUpdateTemplate = true;
    mockFetch({data: {brands: [{name: "test-role", brandName: "test-role", id: "test-role"}, {name: "all", brandName: "all", id: "all"}]}});
    const mockStore = {
      content: {metadata: {...MODALSMETA, ...formFieldMeta}},
      user: {profile}
    };
    store = testStore(mockStore);
    wrapper = mount(<Provider store={store}><CreateUserTemplate data={data}/></Provider>);
    wrapper.find("form.h-100.px-2.pl-0").at(0).simulate('submit');
  })
});
