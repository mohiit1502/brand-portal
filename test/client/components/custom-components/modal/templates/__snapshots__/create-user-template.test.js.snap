// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateUserTemplate test container should render the CreateUserTemplate successfully 1`] = `
<div
  className="modal show create-user-modal"
  id="singletonModal"
  role="dialog"
  tabIndex="-1"
>
  <div
    className="modal-dialog modal-dialog-centered modal-lg"
    role="document"
  >
    <div
      className="modal-content"
    >
      <div
        className="modal-header align-items-center"
      >
        Invite a New User
        <button
          aria-label="Close"
          className="close text-white"
          onClick={[Function]}
          type="button"
        >
          <span
            aria-hidden="true"
          >
            Ã—
          </span>
        </button>
      </div>
      <div
        className="modal-body text-left loader"
      >
        <div
          className="text-secondary font-size-14 mb-2 pl-2"
        >
          Select the type of user you are inviting
        </div>
        <form
          className="h-100 px-2 pl-0"
          onSubmit={[Function]}
        >
          <div
            className="form-row userType"
          >
            <div
              className=" col"
            >
              <div
                className="form-group custom-input-form-group form-group-radio"
              >
                <div
                  className="btn btn-sm radio-btn-box  p-0 active"
                >
                  <input
                    checked={true}
                    className="custom-control-input"
                    id="newuser-userType-custom-input-1"
                    name="newuser-userType-custom-input"
                    onChange={[Function]}
                    type="radio"
                    value="Internal"
                  />
                  <label
                    className="radio-label m-0 p-0"
                    htmlFor="newuser-userType-custom-input-1"
                  >
                    Internal
                  </label>
                </div>
                <div
                  className="btn btn-sm radio-btn-box  p-0 inactive"
                >
                  <input
                    checked={false}
                    className="custom-control-input"
                    id="newuser-userType-custom-input-2"
                    name="newuser-userType-custom-input"
                    onChange={[Function]}
                    type="radio"
                    value="ThirdParty"
                  />
                  <label
                    className="radio-label m-0 p-0"
                    htmlFor="newuser-userType-custom-input-2"
                  >
                    3rd Party
                  </label>
                </div>
              </div>
            </div>
          </div>
          <div
            className="c-HeaderFormComponent row form-prompt"
          >
            <div
              className="col"
            >
              <p>
                Please complete the following fields to invite a new user.
              </p>
            </div>
          </div>
          <div
            className="form-row fname-lname"
          >
            <div
              className=" col-6"
            >
              <div
                className="form-group custom-input-form-group form-group-text mb-3
        "
                style={
                  Object {
                    "position": "static",
                  }
                }
              >
                <input
                  className="form-control form-control-firstName custom-input-element"
                  disabled={false}
                  id="newuser-firstName-custom-input"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onInvalid={[Function]}
                  required={true}
                  type="text"
                  value=""
                />
                
                <label
                  className="custom-input-label custom-input-label-placeholder"
                  htmlFor="newuser-firstName-custom-input"
                >
                  <div
                    className="label-upper-bg position-absolute w-100 h-50 d-block"
                  />
                  <div
                    className="label-lower-bg position-absolute w-100 h-50 d-block"
                  />
                  <span
                    className="label-text required"
                  >
                     
                    First Name
                     
                  </span>
                </label>
                <small
                  className="form-text custom-input-help-text text-muted"
                  style={
                    Object {
                      "paddingLeft": undefined,
                    }
                  }
                >
                  
                </small>
              </div>
            </div>
            <div
              className=" col-6"
            >
              <div
                className="form-group custom-input-form-group form-group-text mb-3
        "
                style={
                  Object {
                    "position": "static",
                  }
                }
              >
                <input
                  className="form-control form-control-lastName custom-input-element"
                  disabled={false}
                  id="newuser-lastName-custom-input"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onInvalid={[Function]}
                  required={true}
                  type="text"
                  value=""
                />
                
                <label
                  className="custom-input-label custom-input-label-placeholder"
                  htmlFor="newuser-lastName-custom-input"
                >
                  <div
                    className="label-upper-bg position-absolute w-100 h-50 d-block"
                  />
                  <div
                    className="label-lower-bg position-absolute w-100 h-50 d-block"
                  />
                  <span
                    className="label-text required"
                  >
                     
                    Last Name
                     
                  </span>
                </label>
                <small
                  className="form-text custom-input-help-text text-muted"
                  style={
                    Object {
                      "paddingLeft": undefined,
                    }
                  }
                >
                  
                </small>
              </div>
            </div>
          </div>
          <div
            className="form-row contact-details"
          >
            <div
              className=" col-6"
            >
              <div
                className="form-group custom-input-form-group form-group-text mb-3
        "
                style={
                  Object {
                    "position": "static",
                  }
                }
              >
                <input
                  className="form-control form-control-emailId custom-input-element"
                  disabled={false}
                  id="newuser-emailId-custom-input"
                  onChange={[Function]}
                  onInvalid={[Function]}
                  pattern="(([^<>()[\\\\]\\\\\\\\.,;:\\\\s@\\"]+(\\\\.[^<>()[\\\\]\\\\\\\\.,;:\\\\s@\\"]+)*)|(\\".+\\"))@((\\\\[[0-9]{1,3}\\\\.[0-9]{1,3}\\\\.[0-9]{1,3}\\\\.[0-9]{1,3}\\\\])|(([a-zA-Z\\\\-0-9]+\\\\.)+[a-zA-Z]{2,}))"
                  required={true}
                  type="email"
                  value=""
                />
                
                <label
                  className="custom-input-label custom-input-label-placeholder"
                  htmlFor="newuser-emailId-custom-input"
                >
                  <div
                    className="label-upper-bg position-absolute w-100 h-50 d-block"
                  />
                  <div
                    className="label-lower-bg position-absolute w-100 h-50 d-block"
                  />
                  <span
                    className="label-text required"
                  >
                     
                    Email
                     
                  </span>
                </label>
                <small
                  className="form-text custom-input-help-text text-muted"
                  style={
                    Object {
                      "paddingLeft": undefined,
                    }
                  }
                >
                  
                </small>
              </div>
            </div>
            <div
              className=" col-6"
            >
              <div
                className="form-group custom-input-form-group form-group-text mb-0
        "
                style={
                  Object {
                    "position": "static",
                  }
                }
              >
                <input
                  className="form-control form-control-phone custom-input-element"
                  disabled={false}
                  id="newuser-phone-custom-input"
                  maxLength={17}
                  onChange={[Function]}
                  onInvalid={[Function]}
                  pattern="^(\\\\([0-9]{3}\\\\) |[0-9]{3}-)[0-9]{3}-[0-9]{4}$"
                  required={false}
                  type="text"
                  value=""
                />
                
                <label
                  className="custom-input-label custom-input-label-placeholder"
                  htmlFor="newuser-phone-custom-input"
                >
                  <div
                    className="label-upper-bg position-absolute w-100 h-50 d-block"
                  />
                  <div
                    className="label-lower-bg position-absolute w-100 h-50 d-block"
                  />
                  <span
                    className="label-text"
                  >
                     
                    Mobile Number
                     
                  </span>
                </label>
                <small
                  className="form-text custom-input-help-text text-muted"
                  style={
                    Object {
                      "paddingLeft": undefined,
                    }
                  }
                >
                  Optional
                </small>
              </div>
            </div>
          </div>
          <div
            className="form-row role-and-brand"
          >
            <div
              className=" col-6"
            >
              <div
                className="form-group custom-input-form-group custom-select-form-group dropdown  mb-3  field-select-arrow"
              >
                <img
                  alt="i"
                  className="positioned-top-right"
                  data-html="true"
                  data-placement="right"
                  data-toggle="tooltip"
                  src="question.svg"
                  title="<div class=\\"py-2\\" data-reactroot=\\"\\"><b class=\\"position-absolute text-white tooltip-close-button\\">x</b><p class=\\"mt-2 pl-2 text-left font-size-12\\"><b>Admin:</b> Can invite reporter-level users, add new brands, submit claims, and view all submitted claims.<br/><br/><b>Reporter:</b> Can submit claims and see submitted claims for their assigned brands.<br/></p></div>"
                />
                <input
                  autoComplete="off"
                  className="form-control form-control-role custom-input-element"
                  data-toggle="dropdown"
                  disabled={false}
                  id="newuser-role-custom-input"
                  onChange={[Function]}
                  required={true}
                  type="select"
                  value=""
                />
                <label
                  className="custom-input-label custom-input-label-placeholder"
                  htmlFor="newuser-role-custom-input"
                >
                  <div
                    className="label-upper-bg position-absolute w-100 h-50 d-block"
                  />
                  <div
                    className="label-lower-bg position-absolute w-100 h-50 d-block"
                  />
                  <span
                    className="label-text"
                  >
                     
                    Set Role
                     
                  </span>
                </label>
                <div
                  className="dropdown-menu"
                />
                <small
                  className="form-text custom-input-help-text text-muted"
                  style={
                    Object {
                      "paddingLeft": "0.7rem",
                    }
                  }
                >
                  
                </small>
              </div>
            </div>
            <div
              className=" col-6"
            >
              <div
                className="form-group custom-input-form-group custom-multi-select-form-group dropdown   mb-3"
              >
                <input
                  autoComplete="off"
                  className="form-control form-control-brands custom-input-element"
                  data-toggle="dropdown"
                  disabled={false}
                  id="newuser-brands-custom-input"
                  onChange={[Function]}
                  required={true}
                  type="multiselect"
                  value=""
                />
                <label
                  className="custom-input-label custom-input-label-placeholder"
                  htmlFor="newuser-brands-custom-input"
                >
                  <div
                    className="label-upper-bg position-absolute w-100 h-50 d-block"
                  />
                  <div
                    className="label-lower-bg position-absolute w-100 h-50 d-block"
                  />
                  <span
                    className="label-text"
                  >
                     
                    Assign Brand
                     
                  </span>
                </label>
                <small
                  className="form-text custom-input-help-text text-muted"
                  style={
                    Object {
                      "paddingLeft": undefined,
                    }
                  }
                >
                  
                </small>
              </div>
            </div>
          </div>
          <div
            className="c-ButtonsPanel form-row action-footer"
            data-test="button-panel"
          >
            <div
              className="col new-user-button-panel text-right"
              data-test="button-div"
            >
              <button
                className="btn btn-sm cancel-btn text-primary"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                Cancel
              </button>
              <button
                className="btn btn-sm btn-primary submit-btn px-3 ml-3"
                disabled={true}
                type="submit"
              >
                Invite
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;

exports[`CreateUserTemplate test container should render the CreateUserTemplate successfully 2`] = `
<div
  className="modal show create-user-modal"
  id="singletonModal"
  role="dialog"
  tabIndex="-1"
>
  <div
    className="modal-dialog modal-dialog-centered modal-lg"
    role="document"
  >
    <div
      className="modal-content"
    >
      <div
        className="modal-header align-items-center"
      >
        Edit User
        <button
          aria-label="Close"
          className="close text-white"
          onClick={[Function]}
          type="button"
        >
          <span
            aria-hidden="true"
          >
            Ã—
          </span>
        </button>
      </div>
      <div
        className="modal-body text-left loader"
      >
        <div
          className="text-secondary font-size-14 mb-2 pl-2"
        >
          Select the type of user you are inviting
        </div>
        <form
          className="h-100 px-2 pl-0"
          onSubmit={[Function]}
        >
          <div
            className="form-row userType"
          >
            <div
              className=" col"
            >
              <div
                className="form-group custom-input-form-group form-group-radio"
              >
                <div
                  className="btn btn-sm radio-btn-box  p-0 active"
                >
                  <input
                    checked={true}
                    className="custom-control-input"
                    id="newuser-userType-custom-input-1"
                    name="newuser-userType-custom-input"
                    onChange={[Function]}
                    type="radio"
                    value="Internal"
                  />
                  <label
                    className="radio-label m-0 p-0"
                    htmlFor="newuser-userType-custom-input-1"
                  >
                    Internal
                  </label>
                </div>
                <div
                  className="btn btn-sm radio-btn-box  p-0 inactive"
                >
                  <input
                    checked={false}
                    className="custom-control-input"
                    id="newuser-userType-custom-input-2"
                    name="newuser-userType-custom-input"
                    onChange={[Function]}
                    type="radio"
                    value="ThirdParty"
                  />
                  <label
                    className="radio-label m-0 p-0"
                    htmlFor="newuser-userType-custom-input-2"
                  >
                    3rd Party
                  </label>
                </div>
              </div>
            </div>
          </div>
          <div
            className="c-HeaderFormComponent row form-prompt"
          >
            <div
              className="col"
            >
              <p>
                Please complete the following fields to invite a new user.
              </p>
            </div>
          </div>
          <div
            className="form-row fname-lname"
          >
            <div
              className=" col-6"
            >
              <div
                className="form-group custom-input-form-group form-group-text mb-3
        "
                style={
                  Object {
                    "position": "relative",
                  }
                }
              >
                <input
                  className="form-control form-control-firstName custom-input-element"
                  disabled={false}
                  id="newuser-firstName-custom-input"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onInvalid={[Function]}
                  required={true}
                  type="text"
                  value="Test Mohit"
                />
                <label
                  className="custom-input-label"
                  htmlFor="newuser-firstName-custom-input"
                >
                  <div
                    className="label-upper-bg position-absolute w-100 h-50 d-block"
                  />
                  <div
                    className="label-lower-bg position-absolute w-100 h-50 d-block"
                  />
                  <span
                    className="label-text required"
                  >
                     
                    First Name
                     
                  </span>
                </label>
                <small
                  className="form-text custom-input-help-text text-muted"
                  style={
                    Object {
                      "paddingLeft": undefined,
                    }
                  }
                >
                  
                </small>
              </div>
            </div>
            <div
              className=" col-6"
            >
              <div
                className="form-group custom-input-form-group form-group-text mb-3
        "
                style={
                  Object {
                    "position": "relative",
                  }
                }
              >
                <input
                  className="form-control form-control-lastName custom-input-element"
                  disabled={false}
                  id="newuser-lastName-custom-input"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onInvalid={[Function]}
                  required={true}
                  type="text"
                  value="Nagar"
                />
                <label
                  className="custom-input-label"
                  htmlFor="newuser-lastName-custom-input"
                >
                  <div
                    className="label-upper-bg position-absolute w-100 h-50 d-block"
                  />
                  <div
                    className="label-lower-bg position-absolute w-100 h-50 d-block"
                  />
                  <span
                    className="label-text required"
                  >
                     
                    Last Name
                     
                  </span>
                </label>
                <small
                  className="form-text custom-input-help-text text-muted"
                  style={
                    Object {
                      "paddingLeft": undefined,
                    }
                  }
                >
                  
                </small>
              </div>
            </div>
          </div>
          <div
            className="form-row contact-details"
          >
            <div
              className=" col-6"
            >
              <div
                className="form-group custom-input-form-group form-group-text disabled mb-3
        "
                style={
                  Object {
                    "position": "relative",
                  }
                }
              >
                <input
                  className="form-control form-control-emailId custom-input-element"
                  disabled={true}
                  id="newuser-emailId-custom-input"
                  onChange={[Function]}
                  onInvalid={[Function]}
                  pattern="(([^<>()[\\\\]\\\\\\\\.,;:\\\\s@\\"]+(\\\\.[^<>()[\\\\]\\\\\\\\.,;:\\\\s@\\"]+)*)|(\\".+\\"))@((\\\\[[0-9]{1,3}\\\\.[0-9]{1,3}\\\\.[0-9]{1,3}\\\\.[0-9]{1,3}\\\\])|(([a-zA-Z\\\\-0-9]+\\\\.)+[a-zA-Z]{2,}))"
                  required={true}
                  type="email"
                  value="wm.ropro.stg+demo@gmail.com"
                />
                <label
                  className="custom-input-label"
                  htmlFor="newuser-emailId-custom-input"
                >
                  <div
                    className="label-upper-bg position-absolute w-100 h-50 d-block"
                  />
                  <div
                    className="label-lower-bg position-absolute w-100 h-50 d-block"
                  />
                  <span
                    className="label-text required"
                  >
                     
                    Email
                     
                  </span>
                </label>
                <small
                  className="form-text custom-input-help-text text-muted"
                  style={
                    Object {
                      "paddingLeft": undefined,
                    }
                  }
                >
                  
                </small>
              </div>
            </div>
            <div
              className=" col-6"
            >
              <div
                className="form-group custom-input-form-group form-group-text mb-0
        "
                style={
                  Object {
                    "position": "relative",
                  }
                }
              >
                <input
                  className="form-control form-control-phone custom-input-element"
                  disabled={false}
                  id="newuser-phone-custom-input"
                  maxLength={17}
                  onChange={[Function]}
                  onInvalid={[Function]}
                  pattern="^(\\\\([0-9]{3}\\\\) |[0-9]{3}-)[0-9]{3}-[0-9]{4}$"
                  required={false}
                  type="text"
                  value="(234) 567-8901"
                />
                <label
                  className="custom-input-label"
                  htmlFor="newuser-phone-custom-input"
                >
                  <div
                    className="label-upper-bg position-absolute w-100 h-50 d-block"
                  />
                  <div
                    className="label-lower-bg position-absolute w-100 h-50 d-block"
                  />
                  <span
                    className="label-text"
                  >
                     
                    Mobile Number
                     
                  </span>
                </label>
                <small
                  className="form-text custom-input-help-text text-muted"
                  style={
                    Object {
                      "paddingLeft": undefined,
                    }
                  }
                >
                  Optional
                </small>
              </div>
            </div>
          </div>
          <div
            className="form-row role-and-brand"
          >
            <div
              className=" col-6"
            >
              <div
                className="form-group custom-input-form-group custom-select-form-group dropdown  mb-3  field-select-arrow"
              >
                <img
                  alt="i"
                  className="positioned-top-right"
                  data-html="true"
                  data-placement="right"
                  data-toggle="tooltip"
                  src="question.svg"
                  title="<div class=\\"py-2\\" data-reactroot=\\"\\"><b class=\\"position-absolute text-white tooltip-close-button\\">x</b><p class=\\"mt-2 pl-2 text-left font-size-12\\"><b>Admin:</b> Can invite reporter-level users, add new brands, submit claims, and view all submitted claims.<br/><br/><b>Reporter:</b> Can submit claims and see submitted claims for their assigned brands.<br/></p></div>"
                />
                <input
                  autoComplete="off"
                  className="form-control form-control-role custom-input-element"
                  data-toggle="dropdown"
                  disabled={false}
                  id="newuser-role-custom-input"
                  onChange={[Function]}
                  required={true}
                  type="select"
                  value="Super Admin"
                />
                <label
                  className="custom-input-label "
                  htmlFor="newuser-role-custom-input"
                >
                  <div
                    className="label-upper-bg position-absolute w-100 h-50 d-block"
                  />
                  <div
                    className="label-lower-bg position-absolute w-100 h-50 d-block"
                  />
                  <span
                    className="label-text"
                  >
                     
                    Set Role
                     
                  </span>
                </label>
                <div
                  className="dropdown-menu"
                />
                <small
                  className="form-text custom-input-help-text text-muted"
                  style={
                    Object {
                      "paddingLeft": "0.7rem",
                    }
                  }
                >
                  
                </small>
              </div>
            </div>
            <div
              className=" col-6"
            >
              <div
                className="form-group custom-input-form-group custom-multi-select-form-group dropdown   mb-3"
              >
                <input
                  autoComplete="off"
                  className="form-control form-control-brands custom-input-element"
                  data-toggle="dropdown"
                  disabled={false}
                  id="newuser-brands-custom-input"
                  onChange={[Function]}
                  required={true}
                  type="multiselect"
                  value=""
                />
                <label
                  className="custom-input-label custom-input-label-placeholder"
                  htmlFor="newuser-brands-custom-input"
                >
                  <div
                    className="label-upper-bg position-absolute w-100 h-50 d-block"
                  />
                  <div
                    className="label-lower-bg position-absolute w-100 h-50 d-block"
                  />
                  <span
                    className="label-text"
                  >
                     
                    Assign Brand
                     
                  </span>
                </label>
                <small
                  className="form-text custom-input-help-text text-muted"
                  style={
                    Object {
                      "paddingLeft": undefined,
                    }
                  }
                >
                  
                </small>
              </div>
            </div>
          </div>
          <div
            className="c-ButtonsPanel form-row action-footer"
            data-test="button-panel"
          >
            <div
              className="col new-user-button-panel text-right"
              data-test="button-div"
            >
              <button
                className="btn btn-sm cancel-btn text-primary"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                Cancel
              </button>
              <button
                className="btn btn-sm btn-primary submit-btn px-3 ml-3"
                disabled={true}
                type="submit"
              >
                Save
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
